File: data/multilang_comparison/program_cpp_vector/src/func28.cpp
#include "program_cpp_vector.hpp" 
Array* func28(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array7 = func29(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array7->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array7->refC--;
      if(array7->refC == 0) {
         array7->data.clear();
         delete array7;
         DEBUG_FREE(array7->id);
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      std::vector<Array*> data_params0(0);
      params0.data = data_params0;
      Array* array15 = func30(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array15->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array15->refC--;
      if(array15->refC == 0) {
         array15->data.clear();
         delete array15;
         DEBUG_FREE(array15->id);
      }
   }
   Array_param params0;
   params0.size = 0;
   std::vector<Array*> data_params0(0);
   params0.data = data_params0;
   Array* array27 = func31(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array27->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   return array27;
}

File: data/multilang_comparison/program_go/src/func28.go
package main
import "math/rand"
var _ = rand.Intn
func Func28(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array7 *Array = Func29(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array7.RefC--
      if array7.RefC == 0 {
          array7.Data = nil
          array7 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array15 *Array = Func30(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array15.RefC--
      if array15.RefC == 0 {
          array15.Data = nil
          array15 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array27 *Array = Func31(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array27;
}

